src/index.js

```diff
@@ -1,390 +1,603 @@
-        return element.nodeType === 3 // Node.TEXT_NODE
-          ? element.nodeValue
-          : recycleElement(element)
-      })
+      if (
+        name in element &&
+        name !== "list" &&
+        name !== "draggable" &&
+        name !== "spellcheck" &&
+        name !== "translate" &&
+        !isSvg
+      ) {
+        element[name] = nextValue == null ? "" : nextValue
+        if (nullOrFalse) {
+          element.removeAttribute(name)
+        }
+      } else {
+        var ns = isSvg && name !== (name = name.replace(/^xlink:?/, ""))
+        if (ns) {
+          if (nullOrFalse) {
+            element.removeAttributeNS(XLINK_NS, name)
+          } else {
+            element.setAttributeNS(XLINK_NS, name, nextValue)
+          }
+        } else {
+          if (nullOrFalse) {
+            element.removeAttribute(name)
+          } else {
+            element.setAttribute(name, nextValue)
+          }
+        }
+      }
     }
   }
+}
+
+var createElement = function(node, lifecycle, eventProxy, isSvg) {
+  var element =
+    node.type === TEXT_NODE
+      ? document.createTextNode(node.name)
+      : (isSvg = isSvg || node.name === "svg")
+        ? document.createElementNS(SVG_NS, node.name)
+        : document.createElement(node.name)
+
+  var props = node.props
+  if (props.onCreate) {
+    lifecycle.push(function() {
+      props.onCreate(element)
+    })
+  }
 
-  function resolveNode(node) {
-    return typeof node === "function"
-      ? resolveNode(node(globalState, wiredActions))
-      : node != null
-        ? node
-        : ""
+  for (var i = 0, length = node.children.length; i < length; i++) {
+    element.appendChild(
+      createElement(node.children[i], lifecycle, eventProxy, isSvg)
+    )
   }
 
-  function render() {
-    skipRender = !skipRender
+  for (var name in props) {
+    updateProperty(element, name, null, props[name], eventProxy, isSvg)
+  }
 
-    var node = resolveNode(view)
+  return (node.element = element)
+}
 
-    if (container && !skipRender) {
-      rootElement = patch(container, rootElement, oldNode, (oldNode = node))
+var updateElement = function(
+  element,
+  lastProps,
+  nextProps,
+  lifecycle,
+  eventProxy,
+  isSvg,
+  isRecycled
+) {
+  for (var name in merge(lastProps, nextProps)) {
+    if (
+      (name === "value" || name === "checked"
+        ? element[name]
+        : lastProps[name]) !== nextProps[name]
+    ) {
+      updateProperty(
+        element,
+        name,
+        lastProps[name],
+        nextProps[name],
+        eventProxy,
+        isSvg
+      )
     }
+  }
 
-    isRecycling = false
+  var cb = isRecycled ? nextProps.onCreate : nextProps.onUpdate
+  if (cb != null) {
+    lifecycle.push(function() {
+      cb(element, lastProps)
+    })
+  }
+}
 
-    while (lifecycle.length) lifecycle.pop()()
+var removeChildren = function(node) {
+  for (var i = 0, length = node.children.length; i < length; i++) {
```

Comment:
node.children.length in length in variable as menionted in the previous comment

Reply:
Do you mean the length of `lifecycle`? In this case, we can't store its length in a variable because we're mutating the array every time we call `.pop()` on it.